unstructured/ingest/v2/processes/connectors/opensearch.py

```diff
@@ -0,0 +1,117 @@
+heavily on the Elasticsearch connector code, inheriting the functionality as much as possible."""
+
+
+@dataclass
+class OpenSearchAccessConfig(AccessConfig):
+    password: Optional[str] = enhanced_field(default=None, sensitive=True)
+    use_ssl: bool = False
+    verify_certs: bool = False
+    ssl_show_warn: bool = False
+    ca_certs: Optional[str] = None
+    client_cert: Optional[str] = None
+    client_key: Optional[str] = None
+
+
+@dataclass
+class OpenSearchClientInput(EnhancedDataClassJsonMixin):
+    http_auth: Optional[tuple[str, str]] = enhanced_field(sensitive=True, default=None)
+    hosts: Optional[list[str]] = None
+    use_ssl: bool = False
+    verify_certs: bool = False
+    ssl_show_warn: bool = False
+    ca_certs: Optional[str] = None
+    client_cert: Optional[str] = None
+    client_key: Optional[str] = None
+
+
+@dataclass
+class OpenSearchConnectionConfig(ConnectionConfig):
+    hosts: Optional[list[str]] = None
+    username: Optional[str] = None
+    access_config: OpenSearchAccessConfig = enhanced_field(sensitive=True)
+
+    def get_client_kwargs(self) -> dict:
+        # Update auth related fields to conform to what the SDK expects based on the
+        # supported methods:
+        # https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/client/__init__.py
+        client_input = OpenSearchClientInput()
+        if self.hosts:
+            client_input.hosts = self.hosts
+        if self.access_config.use_ssl:
+            client_input.use_ssl = self.access_config.use_ssl
+        if self.access_config.verify_certs:
+            client_input.verify_certs = self.access_config.verify_certs
+        if self.access_config.ssl_show_warn:
+            client_input.ssl_show_warn = self.access_config.ssl_show_warn
+        if self.access_config.ca_certs:
+            client_input.ca_certs = self.access_config.ca_certs
+        if self.access_config.client_cert:
+            client_input.client_cert = self.access_config.client_cert
+        if self.access_config.client_key:
+            client_input.client_key = self.access_config.client_key
+        if self.username and self.access_config.password:
+            client_input.http_auth = (self.username, self.access_config.password)
+        logger.debug(
+            f"OpenSearch client inputs mapped to: {client_input.to_dict(redact_sensitive=True)}"
+        )
+        client_kwargs = client_input.to_dict(redact_sensitive=False)
+        client_kwargs = {k: v for k, v in client_kwargs.items() if v is not None}
+        return client_kwargs
+
+    @DestinationConnectionError.wrap
+    @requires_dependencies(["opensearchpy"], extras="opensearch")
+    def get_client(self) -> "OpenSearch":
+        from opensearchpy import OpenSearch
+
+        return OpenSearch(**self.get_client_kwargs())
+
+
+@dataclass
+class OpenSearchUploader(ElasticsearchUploader):
+    connection_config: OpenSearchConnectionConfig
```

Comment:
Seeing this in my IDE: `Non-default argument(s) follows default argument(s) defined in 'ElasticsearchUploader' `

Reply:
Also missing the connector type being set which I think might be causing the above problem.

Reply:
I couldn't seem to reproduce the error above. Even with python 9 . 
I assume its some sort of warning in your ide?
Do you see the same for the elasticsearch connector which has a default arg above non-default?

Also I did add connector_type now. Thought it might pick it up from the inheritance. But it picks up the wrong one. So thanks.